packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -202,10 +202,27 @@ class RenderableTiledMap {
 
   /// This returns an object group fetch by name from a given layer.
   /// Use this to add custom behaviour to special objects and groups.
+  @Deprecated('This method is deprecated. Use the getLayer() method instead.')
   ObjectGroup getObjectGroupFromLayer(String name) {
     final g = map.layers.firstWhere((layer) {
       return layer is ObjectGroup && layer.name == name;
     });
     return g as ObjectGroup;
   }
+
+  /// Returns a layer of type [T] with given [name] from all the layers
+  /// of this map. If no such layer is found, null is returned.
+  T? getLayer<T extends Layer>(String name) {
+    T? foundLayer;
+
+    for (var i = 0; i < map.layers.length; ++i) {
+      final layer = map.layers.elementAt(i);
+      if (layer is T && layer.name == name) {
+        foundLayer = layer;
+        break;
+      }
+    }
+
+    return foundLayer;
+  }
```

Comment:
```suggestion
  T? getLayer<T extends Layer>(String name) {
    T? foundLayer;

    for (final layer in map.layers.length {
      if (layer is T && layer.name == name) {
        foundLayer = layer;
        break;
      }
    }

    return foundLayer;
  }
```

Reply:
Thanks @spydon! I wasn't even aware of `firstWhereOrNull` 😄